sdk/core/core-rest-pipeline/test/multipartPolicy.spec.ts

```diff
@@ -0,0 +1,336 @@
+    url: "https://example.com",
+    method: "POST",
+    ...requestOptions,
+  });
+  const policy = multipartPolicy();
+
+  const successResponse: PipelineResponse = {
+    headers: createHttpHeaders(),
+    request,
+    status: 200,
+  };
+  const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+  next.resolves(successResponse);
+
+  await policy.sendRequest(request, next);
+  return request;
+}
+
+function assertUint8ArraySame(actual: Uint8Array, expected: Uint8Array, message?: string): void {
+  assert.sameOrderedMembers([...actual], [...expected], message);
+}
+
+async function assertBodyMatches(
+  actual: RequestBodyType | undefined,
+  expected: Uint8Array
+): Promise<void> {
+  if (!actual) {
+    assert.fail("Expected a request body");
+  }
+
+  if (isWebReadableStream(actual)) {
+    const actualBytes = new Uint8Array(await new Response(actual).arrayBuffer());
+    assertUint8ArraySame(actualBytes, expected, "body does not match");
+  } else if (isNodeReadableStream(actual)) {
+    const buffers: Buffer[] = [];
+    for await (const buffer of actual) {
+      buffers.push(buffer as Buffer);
+    }
+
+    const actualBytes = new Uint8Array(Buffer.concat(buffers));
+    assertUint8ArraySame(actualBytes, expected, "body does not match");
+  } else {
+    assert.fail(`Requst body of unexpected type: ${actual.toString()}`);
+  }
+}
+
+describe("multipartPolicy", function () {
+  it("passes through when request body is not MultipartRequestBody", async function () {
+    const request = createPipelineRequest({
+      url: "https://example.com",
+      method: "POST",
+      headers: createHttpHeaders({ "content-type": "multipart/form-data" }),
+    });
+
+    const originalRequest = createPipelineRequest({
+      url: "https://example.com",
+      method: "POST",
+      headers: createHttpHeaders({ "content-type": "multipart/form-data" }),
+    });
+
+    originalRequest.requestId = request.requestId;
+
+    const policy = multipartPolicy();
+
+    const successResponse: PipelineResponse = {
+      headers: createHttpHeaders(),
+      request,
+      status: 200,
+    };
+    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.resolves(successResponse);
+
+    await policy.sendRequest(request, next);
+
+    assert.deepStrictEqual(
+      request,
+      originalRequest,
+      "multipartPolicy touched a request that is not multipart"
+    );
+  });
+
+  describe("content-type request header", async function () {
+    it("header is populated to multipart/mixed when not set", async function () {
+      const request = await performRequest({
+        headers: createHttpHeaders({}),
+        body: {
+          parts: [],
+        },
+      });
+
+      assert.ok(request.headers.has("content-type"), "content-type header expected");
+      assert.match(
+        request.headers.get("content-type")!,
+        /multipart\/mixed; boundary=[0-9a-zA-Z'()+,-./:=?]+/,
```

Comment:
maybe the policy itself should be testing that custom boundaries don't violate the spec here? I don't think I saw code ot validate that

Reply:
agree, added validation